src/ProductPage.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState, useEffect } from 'react';
+import css from './ProductPage.module.css';
+
+interface Product {
+  category: {
+    id: number,
+    name: string,
+    image: string
+  };
+  creationAt: string;
+  description: string;
+  id: number;
+  images: string[];
+  price: number;
+  title: string;
+  updatedAt:  string;
+}
+
+const ProductPage: React.FC = () => {
+  const [products, setProducts] = useState<Product[]>([]);
+  const [searchTerm, setSearchTerm] = useState<string>('');
+  const [sortBy, setSortBy] = useState<string>('');
+  const [filterOptions, setFilterOptions] = useState<{ new: boolean; old: boolean }>({
+    new: false,
+    old: false
+  });
+
+  useEffect(() => {
+    fetchProducts();
+  }, []);
+
+  const fetchProducts = async () => {
+    // Симуляція отримання даних з сервера
```

Comment:
You should write comments in English. But in the future, you should read Clean Code book or some alternatives that you can find.